dev/io.openliberty.microprofile6.internal_fat/fat/src/io/openliberty/microprofile6/internal/test/suite/MPCompatibilityTestUtils.java

```diff
@@ -0,0 +1,131 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile6.internal.test.suite;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+
+import com.ibm.websphere.simplicity.PropertiesAsset;
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions;
+
+import componenttest.topology.impl.LibertyServer;
+import io.openliberty.microprofile6.internal.test.helloworld.HelloWorldApplication;
+import io.openliberty.microprofile6.internal.test.helloworld.basic.BasicHelloWorldBean;
+import io.openliberty.microprofile6.internal.test.helloworld.config.ConfiguredHelloWorldBean;
+
+public class MPCompatibilityTestUtils {
+
+    private static final String APP_NAME = "helloworld";
+
+    protected static final String MESSAGE = BasicHelloWorldBean.MESSAGE;
+
+    protected static void setUp(LibertyServer server) throws Exception {
+        setUp(server, true);
+    }
+
+    protected static void setUp(LibertyServer server, boolean startServer) throws Exception {
+
+        PropertiesAsset config = new PropertiesAsset().addProperty("message", MESSAGE);
+
+        WebArchive war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                                   .addPackage(HelloWorldApplication.class.getPackage())
+                                   .addPackage(ConfiguredHelloWorldBean.class.getPackage())
+                                   .addAsResource(config, "META-INF/microprofile-config.properties");
+
+        ShrinkHelper.exportDropinAppToServer(server, war, DeployOptions.SERVER_ONLY);
+
+        if (startServer) {
+            server.startServer();
+        }
+    }
+
+    protected static void cleanUp(LibertyServer server, String... ignoredFailuresRegExps) throws Exception {
+        if ((server != null) && (server.isStarted())) {
+            server.stopServer(ignoredFailuresRegExps);
+        }
+    }
+
+    protected static void validateMessagesAndStop(LibertyServer server, String... expectedFailuresRegExps) throws Exception {
+        if (server != null) {
+            for (String msg : expectedFailuresRegExps) {
+                assertServerMessage(server, msg);
+            }
+            if (server.isStarted()) {
+                server.stopServer(expectedFailuresRegExps);
+            }
+        } else {
+            fail("Server was null");
+        }
+    }
```

Comment:
This will not stop the server if the messages were not found.

Reply:
Ah, but you're still calling cleanup in the `AfterClass` method, so you won't leave it around.